test/backend/models.batch.test.js

```diff
@@ -109,6 +135,23 @@ describe('Report', function() {
     });
   });
 
+  it('should not give same reports to different users', function(done) {
+    var batches = [];
+    createReports(function() {
+      async.filter([user._id, user2._id], function(user, cb) {
```

Comment:
`filter` is not what you want here... either use `each` like this, or `map` and move batches out of the parent scope.